subsys/power/policy/policy_residency.c

```diff
@@ -15,78 +15,57 @@ LOG_MODULE_DECLARE(power);
 
 #define SECS_TO_TICKS		CONFIG_SYS_CLOCK_TICKS_PER_SEC
 
-#if !(defined(CONFIG_SYS_POWER_STATE_CPU_LPS_SUPPORTED) || \
-		defined(CONFIG_SYS_POWER_STATE_CPU_LPS_1_SUPPORTED) || \
-		defined(CONFIG_SYS_POWER_STATE_CPU_LPS_2_SUPPORTED) || \
-		defined(CONFIG_SYS_POWER_STATE_DEEP_SLEEP_SUPPORTED) || \
-		defined(CONFIG_SYS_POWER_STATE_DEEP_SLEEP_1_SUPPORTED) || \
-		defined(CONFIG_SYS_POWER_STATE_DEEP_SLEEP_2_SUPPORTED))
-#error "Enable Low Power States at SoC Level"
-#endif
-
-struct sys_pm_policy {
-	enum power_states pm_state;
-	int sys_state;
-	int min_residency;
-};
-
 /* PM Policy based on SoC/Platform residency requirements */
-static struct sys_pm_policy pm_policy[] = {
+static const unsigned int pm_min_residency[] = {
 #ifdef CONFIG_SYS_POWER_STATE_CPU_LPS_SUPPORTED
-	{SYS_POWER_STATE_CPU_LPS, SYS_PM_LOW_POWER_STATE,
-			CONFIG_PM_LPS_MIN_RES * SECS_TO_TICKS},
+	CONFIG_PM_LPS_MIN_RES * SECS_TO_TICKS,
 #endif
 
 #ifdef CONFIG_SYS_POWER_STATE_CPU_LPS_1_SUPPORTED
-	{SYS_POWER_STATE_CPU_LPS_1, SYS_PM_LOW_POWER_STATE,
-			CONFIG_PM_LPS_1_MIN_RES * SECS_TO_TICKS},
+	CONFIG_PM_LPS_1_MIN_RES * SECS_TO_TICKS,
 #endif
 
 #ifdef CONFIG_SYS_POWER_STATE_CPU_LPS_2_SUPPORTED
-	{SYS_POWER_STATE_CPU_LPS_2, SYS_PM_LOW_POWER_STATE,
-			CONFIG_PM_LPS_2_MIN_RES * SECS_TO_TICKS},
+	CONFIG_PM_LPS_2_MIN_RES * SECS_TO_TICKS,
 #endif
 
 #ifdef CONFIG_SYS_POWER_STATE_DEEP_SLEEP_SUPPORTED
-	{SYS_POWER_STATE_DEEP_SLEEP, SYS_PM_DEEP_SLEEP,
-			CONFIG_PM_DEEP_SLEEP_MIN_RES * SECS_TO_TICKS},
+	CONFIG_PM_DEEP_SLEEP_MIN_RES * SECS_TO_TICKS,
 #endif
 
 #ifdef CONFIG_SYS_POWER_STATE_DEEP_SLEEP_1_SUPPORTED
-	{SYS_POWER_STATE_DEEP_SLEEP_1, SYS_PM_DEEP_SLEEP,
-			CONFIG_PM_DEEP_SLEEP_1_MIN_RES * SECS_TO_TICKS},
+	CONFIG_PM_DEEP_SLEEP_1_MIN_RES * SECS_TO_TICKS,
 #endif
 
 #ifdef CONFIG_SYS_POWER_STATE_DEEP_SLEEP_2_SUPPORTED
-	{SYS_POWER_STATE_DEEP_SLEEP_2, SYS_PM_DEEP_SLEEP,
-			CONFIG_PM_DEEP_SLEEP_2_MIN_RES * SECS_TO_TICKS},
+	CONFIG_PM_DEEP_SLEEP_2_MIN_RES * SECS_TO_TICKS,
 #endif
 };
 
-int sys_pm_policy_next_state(s32_t ticks, enum power_states *pm_state)
+enum power_states sys_pm_policy_next_state(s32_t ticks)
 {
 	int i;
```

Comment:
It's probably better to change `int` to `enum power_states`.

Reply:
No. The `int i` is used as an index to array.